src/services/deduplicationService.ts

```diff
@@ -113,6 +116,64 @@ export function clubDuplicatesByTime(dupes: DuplicateFiles[]) {
     return result;
 }
 
+export function clubDuplicatesBySameFileHashes(dupes: DuplicateFiles[]) {
+    const result: DuplicateFiles[] = [];
+
+    for (const dupe of dupes) {
+        let files: EnteFile[] = [];
+
+        const filteredFiles = dupe.files.filter((file) => {
+            return hasFileHash(file.metadata);
+        });
+
+        if (filteredFiles.length <= 1) {
+            continue;
+        }
+
+        const dupesSortedByFileHash = filteredFiles.map((file) => {
```

Comment:
calling this `dupesSortedByFileHash` is a little weird as we have not sorted them  till now, lets's chain the sort call with this if we want to do so